cmd/list-images/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/deislabs/duffle/pkg/duffle/manifest"
+	"github.com/ghodss/yaml"
+	"github.com/pivotal/image-relocation/pkg/image"
+	"github.com/projectriff/cnab-k8s-installer-base/pkg/apis/kab/v1alpha1"
+	"github.com/projectriff/cnab-k8s-installer-base/pkg/scan"
+)
+
+func main() {
+	bundlePath, manifestPath, err := verifyCommandLineArgs(os.Args)
+	if err != nil {
+		fmt.Printf("error validating arguments %v\n", err)
+		os.Exit(1)
+	}
+
+	mfst := &manifest.Manifest{}
+	err = unmarshallFile(bundlePath, mfst)
+	if err != nil {
+		os.Exit(1)
+	}
+	riffMfst := &v1alpha1.Manifest{}
+	err = unmarshallFile(manifestPath, riffMfst)
+
+	images, err := scanForImages(riffMfst)
+	if err != nil {
+		fmt.Printf("error scanning images %v\n", err)
+		os.Exit(1)
+	}
+	mfst.Images = map[string]bundle.Image{}
+	for _, img := range images {
+		name, err := image.NewName(img)
+		if err != nil {
+			fmt.Printf("err %v\n", err)
+		}
+		n := strings.ReplaceAll(name.Path(), "/", "_")
+		bunImg := bundle.Image{}
+		bunImg.Image = name.String()
+		bunImg.Digest = name.Digest().String()
+		mfst.Images[n] = bunImg
```

Comment:
This will overwrite an existing entry if the paths of distinct images are identical (e.g. "x.com/p" and "y.com/p").